src/UNIV2LPOracle.sol

```diff
@@ -264,14 +294,67 @@ contract UNIV2LPOracle {
         );
     }
 
-    function poke() external stoppable {
-        require(pass(), "UNIV2LPOracle/not-passed");
-        uint128 val = seek();
-        require(val != 0, "UNIV2LPOracle/invalid-price");
-        cur = nxt;
-        nxt = Feed(val, 1);
-        zzz = uint64(block.timestamp);
-        emit Value(cur.val, nxt.val);
+    function poke() external {
+
+        // Ensure a single SLOAD while avoiding solc's excessive bitmasking bureaucracy.
+        uint256 _zph;
+        uint256 _hop;
+        {
+            uint256 _stopped;  // block-scoping _stopped here saves a little gas
+            assembly {
```

Comment:
I like this optimization generally. I wonder if it would be better to declare a local variable in here `let _slot := sload(1)` and use that for masking assignments instead of `_zph`, since it feels a little deceptive to use it this way.

Reply:
Well, you have to be careful...things that make the code more readable sometimes come at the sacrifice of gas :). Here, fortunately, what you suggest does not increase gas costs, so why not...pushing the change directly.